v0/destinations/user/utils.js

```diff
@@ -0,0 +1,617 @@
+ * ref: https://apidocs.user.com/companies/get-company-by-id.html
+ * @param {*} message
+ * @param {*} destination
+ * @returns
+ */
+const getCompanyByCustomId = async (message, destination) => {
+  const { Config } = destination;
+  const { appSubdomain, apiKey } = Config;
+  const companyCustomId = getFieldValueFromMessage(message, "groupId");
+  let { endpoint } = CONFIG_CATEGORIES.GET_COMPANY_BY_COMPANY_CUSTOM_ID;
+  endpoint = prepareUrl(endpoint, appSubdomain).replace(
+    "<company_custom_id>",
+    companyCustomId
+  );
+
+  const requestOptions = {
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Token ${apiKey}`,
+      Accept: "*/*;version=2"
+    }
+  };
+
+  const response = await httpGET(endpoint, requestOptions);
+  const processedUserResponse = processAxiosResponse(response);
+  if (processedUserResponse.status === 200) {
+    return processedUserResponse.response;
+  }
+
+  return null;
+};
+
+/**
+ * lookup logic to find user
+ * priority : userKey(eternalId) -> lookup field(integrationObj) -> userId -> default lookup(email)
+ * @param {*} message
+ * @param {*} destination
+ * @returns
+ */
+const retrieveUserFromLookup = async (message, destination) => {
+  const { Config } = destination;
+  const { appSubdomain, apiKey } = Config;
+  const userKey = getDestinationExternalID(message, "userKey");
+  if (isDefinedAndNotNullAndNotEmpty(userKey)) {
+    return getUserByUserKey(apiKey, userKey, appSubdomain);
+  }
+
+  const integrationsObj = getIntegrationsObj(message, "user");
+  if (integrationsObj && integrationsObj.lookup) {
+    const { lookup: lookupField } = integrationsObj;
+    const lookupFieldValue = getFieldValueFromMessage(message, lookupField);
+
+    if (lookupField === "email") {
+      return getUserByEmail(apiKey, lookupFieldValue, appSubdomain);
+    }
+
+    if (lookupField === "phone") {
+      return getUserByPhoneNumber(apiKey, lookupFieldValue, appSubdomain);
+    }
+    throw new CustomError(
+      `lookup field : ${lookupField} is not supported for this destination`,
+      400
+    );
+  } else {
+    const userId = getValueFromMessage(message, "userId");
+    if (userId) {
+      return getUserByCustomId(message, destination);
+    }
+    const email = getFieldValueFromMessage(message, "email");
+    if (isDefinedAndNotNullAndNotEmpty(email)) {
+      return getUserByEmail(apiKey, email, appSubdomain);
+    }
+    throw new CustomError("default lookup field : email value is empty", 400);
+  }
+};
+
+/**
+ * Returns 'Create User' payload
+ * @param {*} message
+ * @param {*} destination
+ * @returns
+ */
+const createUserPayloadBuilder = (message, destination) => {
+  let payload = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.CREATE_USER.name]
+  );
+  const { userAttributesMapping } = destination.Config;
+  const userAttributes = getUserDestAttributes(userAttributesMapping, message);
```

Comment:
Name it as `getCustomUserAttributes` and the variable name as `customUserAttributes`